src/domains/chart/hooks/useAlarmBadge.js

```diff
@@ -0,0 +1,30 @@
+import { useRef } from "react"
+import { useToggle } from "react-use"
+
+export default () => {
+  const [isRendered, toggleIsRendered] = useToggle(false)
+
+  const ref = useRef(null)
+
+  const updatePosition = (alarm, g, position) => {
+    if (!alarm) {
```

Comment:
If we rename the alarm parameter to something more general like `visible`.
Then the entire `useAlarmBadge` will be more general to `useBadge`